aten/src/ATen/native/ScatterGather.cpp

```diff
@@ -0,0 +1,82 @@
+#include <tuple>
+#include <algorithm>
+#include <ATen/ATen.h>
+#include <c10/core/WrapDimMinimal.h>
+
+namespace at { namespace native {
+
+Tensor & gather_out(Tensor & result, const Tensor & self, int64_t dim, const Tensor & index, bool sparse_grad) {
```

Comment:
I don't really understand the process used to move these functions -- e.g. why is masked_scatter still in Indexing?  Did you think the file was getting too big?

Reply:
Because I think it would be cleaner to put everything (definition, backward, broadcasting) about `torch.scatter`, `torch.gather`, `torch.scatter_add` in the same file. I don't mind putting it back to `Indexing.cpp`. For `masked_scatter`, I think it is better to stay together with `masked_fill`, and its semantics is more similar to advanced indexing, so better stay at `Indexing.cpp`. But I am also OK with moving to `ScatterGather.cpp`